### PR TITLE
Design docs

### PR DESCRIPTION
**Pull request by eamsden**

### CODE REVIEW

File: docs/heap.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Heaps and persistence
+
+However, there are several reasons why we may wish to copy a noun off the stack and reference it during a computation.
+Copying a large noun of the stack and maintaining a reference instead prevents repeated copying up the stack.
+Fixing a location for a noun enables pointer equality to short-circuit equality checks.
+Snapshotting and paging requires a fixed location in memory for a noun, unperturbed by computation.
+Finally, the ability to attach metadata (hints, compiled code, reference counts, etc) to nouns is extremely helpful for runtime implementation, but costly if provided for every noun.
+Allowing metadata to be attached to heap entries
```
Comment (joemfb) [inline docs/heap.md:8]: This got cut off.

File: docs/llvm.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+| `ir[0 0 [9 b c]]`       | `ir[0 1 c]; sub; axe[b]; lnt`                                                
+| `ir[0 1 [9 b c]]`       | `puh[1]; ir[0 1 c]; sub; axe[b]; lnk; pop`                                   
+| `ir[1 1 [9 b c]]`       | `puh[2]; sav[1]; ir[0 1 c]; sub; axe[b]; lnk; reo[1]; pop`                   
+| `ir[0 0 [10 [b c] d]]`  | `puh[1]; ir[1 1 d]; put[0]; ir[0 1 c]; reo[0]; edt[b]; pop; don`             
+| `ir[0 1 [10 [b c] d]]`  | `puh[1]; ir[1 1 d]; put[0]; ir[0 1 c]; reo[0]; edt[b]; pop;`                 
+| `ir[1 1 [10 [b c] d]]`  | `puh[2]; sav[1]; ir[0 1 d]; put[0]; reo[1]; ir[0 1 c]; reo[0]; edt[b]; pop   
+| `ir[s t [11 [b c] d]]`  | `ir[1 1 b]; hnd[b]; ir[s t d]`                                               
+| `ir[s t [11 b c]]       | `hns[b]; ir[s t c]`                                                          
+| `ir[0 0 [12 b c]]       | `puh[1]; ir[1 1 b]; put[0]; ir[0 1 c]; cel[0]; spy; pop; don                 
+| `ir[s 1 [12 b c]]       | `puh[1]; ir[1 1 b]; put[0]; ir[s 1 c]; cel[0]; spy; pop
```
Comment (joemfb) [inline docs/llvm.md:88]: Missing some closing `` ` ``s here.

File: docs/subject-knowledge.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+If the formula isn't known, then the result is `[%gues ~]`, though the first subformula may still be analyzed for code-generation purposes.
+
+### 3
+SKA of 3 recursively analyzes the subformula, then examines what is known of the structure of the noun. Specifically, for the %know, %bets, or %dice cases, it is known whether the result is a cell or an atom, and thus
+the analysis can produce `[%know 0]` for a known cell or `[%know 1]` for a known atom. For the `[%gues ~]` result, however, we must produce `[%dice ~]`
+
+### 4
+SKA of 4 *intentionally* does not produce the obvious result, in order to short-circuit full analysis of functions which should be jetted.
+If the result of the subformula is known to be a cell, we can produce knowledge of a crash.
+Otherwise the result is `[%dice ~]` whether or not the atom result of the subformula is known.
```
Comment (joemfb) [inline docs/subject-knowledge.md:70]: It can only be `%dice` if the subformula is known or dice, right?

**Review by joemfb — CHANGES_REQUESTED** [review]
These look great! I found a couple things:

File: docs/heap.md (hunk 1 above)
Comment (eamsden) [reply docs/heap.md]: Thanks, fixed

File: docs/llvm.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+| `ir[0 0 [9 b c]]`       | `ir[0 1 c]; sub; axe[b]; lnt`                                                
+| `ir[0 1 [9 b c]]`       | `puh[1]; ir[0 1 c]; sub; axe[b]; lnk; pop`                                   
+| `ir[1 1 [9 b c]]`       | `puh[2]; sav[1]; ir[0 1 c]; sub; axe[b]; lnk; reo[1]; pop`                   
+| `ir[0 0 [10 [b c] d]]`  | `puh[1]; ir[1 1 d]; put[0]; ir[0 1 c]; reo[0]; edt[b]; pop; don`             
+| `ir[0 1 [10 [b c] d]]`  | `puh[1]; ir[1 1 d]; put[0]; ir[0 1 c]; reo[0]; edt[b]; pop;`                 
+| `ir[1 1 [10 [b c] d]]`  | `puh[2]; sav[1]; ir[0 1 d]; put[0]; reo[1]; ir[0 1 c]; reo[0]; edt[b]; pop`
+| `ir[s t [11 [b c] d]]`  | `ir[1 1 b]; hnd[b]; ir[s t d]`                                               
+| `ir[s t [11 b c]]`      | `hns[b]; ir[s t c]`                                                          
+| `ir[0 0 [12 b c]]`      | `puh[1]; ir[1 1 b]; put[0]; ir[0 1 c]; cel[0]; spy; pop; don`  
+| `ir[s 1 [12 b c]]       | `puh[1]; ir[1 1 b]; put[0]; ir[s 1 c]; cel[0]; spy; pop`
```
Comment (joemfb) [inline docs/llvm.md:88]: One more backtick here.

File: docs/subject-knowledge.md (hunk 3 above)
Comment (joemfb) [reply docs/subject-knowledge.md:75]: Thanks, this makes sense.

**Pull request merged**